app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala

```diff
@@ -111,22 +114,17 @@ object SubscriptionFieldsService {
     val empty = Map.empty[FieldName, ErrorMessage]
   }
 
-  // True - passed
-  def validateAgainstRule(rule: ValidationRule, value: String): Boolean = rule match {
-    case RegexValidationRule(regex) => value.matches(regex)
-  }
-
   // True - passed
   def validateAgainstGroup(group: ValidationGroup, value: String): Boolean = {
-    group.rules.foldLeft(true)((acc, rule) => (acc && validateAgainstRule(rule, value)))
+    group.rules.foldLeft(true)((acc, rule) => (acc && rule.validate(value)))
   }
 
   // Some is Some(error)
   def validateAgainstDefinition(fieldDefinition: FieldDefinition, value: String): Option[FieldError] = {
-    fieldDefinition.validation.flatMap(group => if (validateAgainstGroup(group, value)) None else Some((fieldDefinition.name, group.errorMessage)))
+    fieldDefinition.validation .flatMap(group => if (validateAgainstGroup(group, value)) None else Some((fieldDefinition.name, group.errorMessage)))
   }
 
-  def validate(fieldDefinitions: NonEmptyList[FieldDefinition], fields: Fields): FieldErrorMap =
+  def validateAgainstValidationRules(fieldDefinitions: NonEmptyList[FieldDefinition], fields: Fields): FieldErrorMap =
     fieldDefinitions
       .map(fd => validateAgainstDefinition(fd, fields.get(fd.name).getOrElse("")))
```

Comment:
Looks like this could be shortened to:

.map(fd => validateAgainstDefinition(fd, fields.getOrElse(fd.name, "")))

Reply:
Nice spot